app/(default)/webstories/[slug]/page.tsx

```diff
@@ -32,92 +46,88 @@ const Index: React.FC = () => {
     }
   }, []);
 
-  const preloadImages = async (images: string[]) => {
-    const loadedImages = await Promise.all(
-      images.map(
-        (url) =>
-          new Promise<PreloadedImage>((resolve) => {
-            const img = new (window as any).Image();
-            img.src = url;
-            img.onload = () => resolve({ url, img });
-          })
-      )
-    );
-    setPreloadedImages(loadedImages);
-    setIsPreloading(false);
-  };
-
-  const loadCurrentAndNextImage = (index: number) => {
-    if (storyData) {
-      const imageUrls = [
-        storyData[index]?.imageUrl,
-        storyData[index + 1]?.imageUrl,
-      ].filter(Boolean) as string[];
-      preloadImages(imageUrls);
-    }
-  };
+  // const preloadImages = async (images: (string | StaticImageData)[]) => {
+  //   const loadedImages = await Promise.all(
+  //     images.map(
+  //       (url) =>
+  //         new Promise<PreloadedImage>((resolve) => {
+  //           const img =  new (window as any).Image();;
+  //           img.src = typeof url === "string" ? url : url.src;
+  //           img.onload = () => resolve({ url, img });
+  //         })
+  //     )
+  //   );
+  //   setPreloadedImages(loadedImages);
+  //   setIsPreloading(false);
+  // };
 
-  useEffect(() => {
-    loadCurrentAndNextImage(currentStoryIndex);
-  }, [currentStoryIndex, storyData]);
+  // useEffect(() => {
+  //   if (storyData) {
+  //     const imageUrls = storyData.map((story) => story.imageUrl).filter(Boolean) as (string | StaticImageData)[];
+  //     preloadImages(imageUrls);
+  //   }
+  // }, [storyData]);
 
   const handleClose = () => {
-    window.close();
+    window.location.href = "/webstories";
```

Comment:
why not use next router instead of window.location, as window.location.href will trigger whole window reload ?

Reply:
ok did it!!